Add command block

mroote opened the pull request:
* Adds a block that supports output from any shell commands.

mroote committed:
Add block type to allow shell commands to run

mroote committed:
Parse command arguments, add example to config file

davidscholberg commented on lib/modules/command.go:
```diff
@@ -0,0 +1,30 @@
+		return
+	}
+
+	b.FullText = fmt.Sprintf(fullTextFmt, strings.Replace(string(out), "\n", "", -1))
```
If your intention was to only remove the trailing newline, then this can more easily be accomplished by using the [strings.TrimSpace](https://golang.org/pkg/strings/#TrimSpace) function.

If you want to also remove newlines from multiline output, then I'd recommend replacing those with a space rather than just removing them.

davidscholberg replied:
Also, I forgot to mention that this gives me a compile error:

```
# github.com/mroote/goblocks/lib/modules
lib/modules/command.go:29: undefined: out
```

This appears to be due to the `out` variable being declared in the if block, meaning that it's not available outside the block's scope. You can resolve this by just moving the `exec.Command` call before the if block, which will declare the `out` variable in the proper scope.

davidscholberg commented on lib/modules/configure.go:
```diff
@@ -152,8 +152,14 @@ func getBlockConfigInstance(m map[string]interface{}) (*BlockConfig, error) {
 	}
-	return nil, fmt.Errorf("error: type %s not valid", t)
+
+	return nil, fmt.Errorf("type %s not valid", t)
```
Why was `error: ` removed here?

mroote replied:
The error messages already had an "error:" string so it was getting repeated twice when errors were generated.

davidscholberg replied:
Doh! You're right. Thank you :D

davidscholberg reviewed (changes requested):
I just have a couple small requested changes for the command module.

As for the battery and disk blocks, I like the enhancements that you made, but if it's not too much trouble, can you remove them from this pull request and create separate ones for each? I think that would make the review process go a little smoother.

mroote committed:
Use TrimSpace to remove whitespace from cmd output

mroote commented:
Hi thanks for the comments, I've moved my other changes to different branches and could open PR's for those if you are interested in pulling in the changes.

I've also addressed the comments you left, let me know if there are any other issues you can see.

Thanks

davidscholberg commented:
@mroote Thanks for the updates! Everything looks good.

I'm going to change this pull request to pull into my local command branch rather than master so that I can update the README and release all of this to master at the same time. I'll notify you here when these changes make it to master.

As for you other changes, please feel free to submit pull requests for them. I'd be glad to merge those changes in :D

Thanks again for your contribution!

davidscholberg commented:
@mroote your changes were merged to master in #16.
